### PR TITLE
Code review

### PR DESCRIPTION
**Pull request by dferendo**
## Description

Code review for farming module

## Tasks

- [ ] 

## References

- 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Appropriate labels applied
- [ ] Targeted PR against correct branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://go.dev/blog/godoc).
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/farming/keeper/reward.go [hunk 1]
```diff
@@ -394,6 +399,8 @@ func (k Keeper) AllocateRewards(ctx sdk.Context) error {
 			continue
 		}
 
+		// Should the rewards for all proposals be kept in the same address?
+		// Would rather prefer having a reward address per proposal just in case there is a bug with the maths
```
[the remark this answers is not in the document]
Comment (hallazzang) [reply x/farming/keeper/reward.go:403]: I appreciate your suggestion.
@dongsam @kogisin @dlguddus How do you think about this?

File: x/farming/keeper/staking.go [hunk 2]
```diff
@@ -295,9 +295,7 @@ func (k Keeper) Unstake(ctx sdk.Context, farmerAcc sdk.AccAddress, amount sdk.Co
 			}
 
 			k.DecreaseTotalStakings(ctx, coin.Denom, removedFromStaking)
-		}
-
-		if queuedStaking.Amount.IsPositive() {
+		} else if queuedStaking.Amount.IsPositive() {
```
[the remark this answers is not in the document]
Comment (hallazzang) [reply x/farming/keeper/staking.go:298]: We can delete a `QueuedStaking` when the amount of unstaking coins is greater than the amount of a `QueuedStaking`.
Suppose we have `100atom` staked and `50atom` queued for staking.
When we unstake `70atom`, the result of `queuedStakingCoinAmt - unstakingCoinAmt` would be -20 which is negative, not 0.
We then delete the queued staking(`50atom` in this case) and staking coin amount becomes `80atom`(which is `100atom - -(-20atom)` following formula in the code).

[the remark this answers is not in the document]
Comment (ferendo) [reply x/farming/keeper/staking.go:298]: Yup makes sense

File: x/farming/keeper/reward.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dongsam) [reply x/farming/keeper/reward.go:403]: Thank you for the good suggestion. I think it'll be safer and better to track. 
I think rather, `StakingReserveAcc` is more intuitive when divided by staking denim, but `RewardsReserveAcc` may be different between staking coin denom and farming reward denom, and it is a temporary pool that stays before withdrawing, so a relatively small amount stays for a while.

[the remark this answers is not in the document]
Comment (dongsam) [reply x/farming/keeper/reward.go:403]: I just open a PR https://github.com/tendermint/farming/issues/220 for this, Thanks!

**Comment by dongsam** [comment]
Close the issue because all suggestions have been discussed or resolved. Thank you again.

**Pull request closed without merging**